### PR TITLE
pricing-plans page

### LINKED ISSUE #964: Pricing page

### PR DESCRIPTION
**Pull request by Roman4437**
close #964

### CODE REVIEW

File: src/router/index.ts [hunk 1]
```diff
@@ -22,6 +22,11 @@ const routes: Array<RouteRecordRaw> = [
         path: 'dashboard',
         component: () => import('../pages/admin/dashboard/Dashboard.vue'),
       },
+      {
+        name: 'Payment methods',
```
Comment (asvae) [inline src/router/index.ts:26]:
```suggestion
        name: 'payment-methods',
```

File: src/router/index.ts [hunk 2]
```diff
@@ -22,6 +22,11 @@ const routes: Array<RouteRecordRaw> = [
...
+        path: '/pricing-plans',
+        component: () => import('../pages/pricing-plans/PricingPlans.vue'),
```
Comment (asvae) [inline src/router/index.ts:28]: Let's keep all names consistent (for route, navigation, component name). Please take name from figma.

File: src/pages/pricing-plans/options.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+]
+
+export const pricingPlans: PricingPlans[] = [
+  {
+    title: 'Startup',
+    model: 'Startup',
+    description: 'Optimal for 10+ team size and new startup',
+    price: 39,
+    features: features.map((d, i) => ({ description: d, isAvailable: i < 3 })),
+    class: '!bg-[#F7F9F9]',
```
Comment (asvae) [inline src/pages/pricing-plans/options.ts:33]: Don't we have similar colors in vuestic? (would be ideal to either use existing or have a follow up on this)

File: src/pages/pricing-plans/PricingPlans.vue [hunk 4]
```diff
@@ -0,0 +1,91 @@
+<template>
+  <div class="text-textPrimary">
+    <h1 class="text-[32px] md:text-5xl leading-[110%] md:leading-[56px] font-bold">
+      Choose your plan
+    </h1>
+    <div class="py-4 text-lg leading-[26px]">
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:6]: > leading-[26px]

Let's try to minimize usage of -[123] tailwind classes.

you can use custom class (as page-title) or just use global default

File: src/pages/pricing-plans/PricingPlans.vue [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      If you need more info about our pricing, please check <span class="text-primary underline">Pricing Guidelines</span>.
+    </div>
+    <div class="flex flex-col p-4 bg-backgroundSecondary">
+      <div class="flex justify-center">
+        <div class="space-x-[1px] p-[2px] border rounded">
+          <va-button
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:12]: Let's move this to a separate component (in admin). It does not have to be as intricate as v-ui components, planned for rework anyway.

File: src/pages/pricing-plans/PricingPlans.vue [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      Choose your plan
+    </h1>
+    <div class="py-4 text-lg leading-[26px]">
+      If you need more info about our pricing, please check <span class="text-primary underline">Pricing Guidelines</span>.
+    </div>
+    <div class="flex flex-col p-4 bg-backgroundSecondary">
+      <div class="flex justify-center">
+        <div class="space-x-[1px] p-[2px] border rounded">
+          <va-button 
+            v-for="option in ['Monthly', 'Annual']"
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:13]: You can use boolean for value (isAnnual).

File: src/pages/pricing-plans/PricingPlans.vue [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            class="w-[85.5px]"
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:14]: you can standard tailwind classes or at least rounder numbers.

Maybe also use rem? (not sure if that's correct idea)

File: src/pages/pricing-plans/PricingPlans.vue [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+            color="textPrimary"
+            preset="plain"
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:16]: I suggest not to use preset in this case, you're mostly overriding it already.

File: src/pages/pricing-plans/PricingPlans.vue [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+            :pressedOpacity="0.85"
+            :hoverOpacity="0.85"
+            :style="{ ...durationButtonStyles, ...( planDuration === option && { '--va-0-background-color': '#ECF0F1' } ) }"
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:19]: suggest to move it to `script` - pretty hard to read

File: src/pages/pricing-plans/PricingPlans.vue [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+            @click="planDuration = option"
+          >
+            {{ option }}
+          </va-button>
+        </div>
+      </div>
+      <div class="flex flex-col md:flex-row justify-center items-center space-y-4 md:space-x-6 md:space-y-0 mt-6">
+        <va-card 
+          v-for="plan in pricingPlans"
+          class="flex w-[326px] md:w-[349px] h-fit p-6 rounded-[13px]"
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:29]: It probably should work without relying on specific widths.

File: src/pages/pricing-plans/PricingPlans.vue [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+          :class="plan.class"
+        >
+          <div 
+            class="space-y-8 md:space-y-10"
+            :class="{ '!space-y-10' : plan.model === 'Advanced' }"
+          >
+            <div class="space-y-4 text-center">
+              <h2 class="text-[28px] md:text-[32px] leading-10 font-bold">
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:37]: Let's move this to custom class (similar to `.page-title`)

File: src/pages/pricing-plans/PricingPlans.vue [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+                {{ plan.title }}
+              </h2>
+              <va-badge 
+                v-for="b in plan.badges"
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:41]: It's considered a good practice to use abbreviations sparingly

So would recommend `badge` here

File: src/pages/pricing-plans/PricingPlans.vue [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+                <va-icon v-if="feature.isAvailable" name="mso-check" color="primary" size="20px" />
+                <va-icon v-else name="mso-block" color="backgroundBorder" size="20px" />
+              </div>
+            </div>
+            <div class="flex justify-center">
+              <va-button 
+                :style="selectButtonStyles"
+                @click="selectedPlan = plan.model"
+              >
+                Select
```
Comment (asvae) [inline src/pages/pricing-plans/PricingPlans.vue:73]: We need something to happen visually on plan select.

My suggestion:
* Use success color and disabled for current plan, text should read "Current plan".
* Same style for non selected styles.
* Show notification on select telling you you successfully selected a plan name (you can ask gpt for originality I'm lacking 🤪)

**Review by asvae — CHANGES_REQUESTED** [review]
Thanks! Excellent work on templating.

I think 1 round of fixes should be enough, but let's have one more review just in case.

File: src/pages/pricing-plans/options.ts (hunk 3 above)
Comment (Roman4437) [reply src/pages/pricing-plans/options.ts:33]: there's no such color

File: src/pages/pricing-plans/PricingPlans.vue (hunk 10 above)
Comment (Roman4437) [reply src/pages/pricing-plans/PricingPlans.vue:29]: it doesnt

**Comment by asvae** [comment]
image

This switch should update prices from one month to one year (actually do something).

**Review by asvae — APPROVED** [review]
Great work, thanks!

**Pull request merged**